src/components/icon/js/iconDirective.js

```diff
@@ -182,9 +182,15 @@ function mdIconDirective($mdIcon, $mdTheming, $mdAria ) {
    */
   function postLink(scope, element, attr) {
     $mdTheming(element);
+    var lastIcon = scope.fontIcon;
+    var lastSet = $mdIcon.fontSet(scope.fontSet);
 
     prepareForFontIcon();
 
+    attr.$observe('mdFontIcon', fontIconChanged);
```

Comment:
I did a quick search on the project and came with a lot of places we don't check it
Basically when we use `$observe` we don't check if the attribute has value, only on `slider` and that's because we want to apply different function if there's no attribute